README.md

```diff
@@ -31,14 +69,393 @@ Permissions to allow the user to interact with the support center:
+status:
+  claimed: true
+  conditions:
+  - lastProbeTime: 2019-07-03T16:14:50Z
+    lastTransitioNTime: 2019-07-03T16:14:50Z
+  	message: Attempting to create account
+ 	  reason: Creating
+ 	  status: "True"
+ 	  type: Creating
+  - lastProbeTime: 2019-07-03T16:18:55Z
+    lastTransitioNTime: 2019-07-03T16:18:55Z
+    message: Account pending AWS limits verification
+    reason: PendingVerification
+    status: "True"
+    type: PendingVerification
+  - lastProbeTime: 2019-07-05T13:19:32Z
+    lastTransitioNTime: 2019-07-05T13:19:32Z
+    message: Account ready to be claimed
+    reason: Ready
+    status: "True"
+    type: Ready
+  state: Ready
+  supportCaseID: case-000000000-muen-2019-000000000000
+```
+
+
+## 2.3. AccountClaim CR
+
+The AccountClaim CR holds the required data for cluster provisioning to build the cluster 
+
+```yaml
+apiVersion: aws.managed.openshift.io/v1alpha1
+kind: AccountClaim
+metadata:
+  creationTimestamp: 2019-07-16T13:52:02Z
+  generation: 2
+  labels:
+    api.openshift.com/id: 00000000000000000000
+    api.openshift.com/name: example-link
+  name: example-link
+  namespace: {NameSpace cluster is being built in}
+  resourceVersion: "54324077"
+  selfLink: /apis/aws.managed.openshift.io/v1alpha1/namespaces/uhc-staging-16toh05i9h8ook5d7ej4al7ejec17rug/accountclaims/razevedo-test2
+  uid: b7113d4a-a8cb-11e9-a2a3-2a2ae2dbcce4
+spec:
+  accountLink: osd-{accountName} (From AccountClaim)
+  aws:
+    regions:
+    - name: us-east-1
+  awsCredentialSecret:
+    name: aws
+    namespace: {NameSpace cluster is being built in}
+  legalEntity:
+    id: 00000000000000
+    name: {Legal Entity Name}
+status:
+  conditions:
+  - lastProbeTime: 2019-07-16T13:52:02Z
+    lastTransitionTime: 2019-07-16T13:52:02Z
+    message: Attempting to claim account
+    reason: AccountClaimed
+    status: "True"
+    type: Unclaimed
+  - lastProbeTime: 2019-07-16T13:52:03Z
+    lastTransitionTime: 2019-07-16T13:52:03Z
+    message: Account claimed by osd-creds-mgmt-fhq2d2
+    reason: AccountClaimed
+    status: "True"
+    type: Claimed
+  state: Ready
+```
+# 3. The controllers
+
+## 3.1. AccountPool Controller
+
+The accountpool-controller is triggered by an accountpool CR or an account CR. It is responsible for generating new account CRs. 
+
+It looks at the accountpool CR *spec.poolSize* and it ensures that the number of unclaimed accounts matchs the number of the poolsize. If the number of unclaimed accounts is less then the poolsize it creates a new account CR for the account-controller to process.
+
+### 3.1.1. Constants and Globals
+```
+emailID = "osd-creds-mgmt"
+```
+
+### 3.1.2. Status
+
+Updates accountPool cr 
+```
+  claimedAccounts: 4
+  poolSize: 3
+  unclaimedAccounts: 3
+```
+
+*claimedAccounts* are any accounts with the `status.Claimed=true`
+*unclaimedAccounts* are any accounts with `status.Claimed=false` and `status.State!="Failed"`
+*poolSize* is the poolsize from the accountPool spec
+
+### 3.1.3. Metrics
```

Comment:
Thinking about this, it may be good to remove the metric definitions from each controllers section and just move it at a generic `metrics` section. The metrics themselves should be agnostic from what controller they are updated in, so a single section of explanation and reference will make it clearer.

Reply:
I agree with this as well. For the next refactor

Reply:
Again lets not talk about the next refactor, will this be part of the constants/global refactor or the refactoring of the EC2 instance status loop or this a separate metrics refactor? Its gets a bit confusing :slightly_smiling_face:

We should give a little more detail here around how the metrics are updated for each controller.